robottelo/ssh.py

```diff
@@ -73,18 +76,28 @@ def _get_connection(
         hostname = settings.server.hostname
     if username is None:
         username = settings.server.ssh_username
-    if key_filename is None:
+    if key_filename is None and password is None:
         key_filename = settings.server.ssh_key
+        # If we don't have key even in settings, read password instead:
+        if key_filename is None:
+            password = settings.server.ssh_password
```

Comment:
```python
if key_filename is None:
    key_filename = settings.server.ssh_key
if password is None:
    password = settings.server.ssh_password
```
The idea here is to initialize variables, either with provided argument values or at least by reading settings. I see you want to prioritize ssh key over a password here, but there's no need in nested checks - at [line 87](https://github.com/SatelliteQE/robottelo/pull/3362/files#diff-8ff1961bf9b83e156d133a872e9b0a56R87) you're prioritizing it anyway